client/network/src/protocol/generic_proto/handler/group.rs

```diff
@@ -304,52 +291,45 @@ impl ProtocolsHandler for NotifsHandler {
 			NotifsHandlerIn::Enable => {
 				self.enabled = EnabledState::Enabled;
 				self.legacy.inject_event(LegacyProtoHandlerIn::Enable);
-				for (handler, _) in &mut self.out_handlers {
+				for handler in &mut self.out_handlers {
 					handler.inject_event(NotifsOutHandlerIn::Enable {
 						initial_message: vec![]
 					});
 				}
 				for num in self.pending_in.drain(..) {
-					self.in_handlers[num].0.inject_event(NotifsInHandlerIn::Accept(vec![]));
+					self.in_handlers[num].inject_event(NotifsInHandlerIn::Accept(vec![]));
 				}
 			},
 			NotifsHandlerIn::Disable => {
 				self.legacy.inject_event(LegacyProtoHandlerIn::Disable);
 				// The notifications protocols start in the disabled state. If we were in the
 				// "Initial" state, then we shouldn't disable the notifications protocols again.
 				if self.enabled != EnabledState::Initial {
-					for (handler, _) in &mut self.out_handlers {
+					for handler in &mut self.out_handlers {
 						handler.inject_event(NotifsOutHandlerIn::Disable);
 					}
 				}
 				self.enabled = EnabledState::Disabled;
 				for num in self.pending_in.drain(..) {
-					self.in_handlers[num].0.inject_event(NotifsInHandlerIn::Refuse);
+					self.in_handlers[num].inject_event(NotifsInHandlerIn::Refuse);
 				}
 			},
 			NotifsHandlerIn::SendLegacy { message } =>
 				self.legacy.inject_event(LegacyProtoHandlerIn::SendCustomMessage { message }),
-			NotifsHandlerIn::SendNotification { message, engine_id, protocol_name } => {
-				for (handler, ngn_id) in &mut self.out_handlers {
+			NotifsHandlerIn::SendNotification { message, encoded_fallback_message, protocol_name } => {
+				for handler in &mut self.out_handlers {
 					if handler.protocol_name() != &protocol_name[..] {
-						break;
+						continue;
```

Comment:
This is a bugfix. This loop tries all the protocols that have been registered. If it can't find any, it uses the legacy substream itself. Because of this `break`, we were using the legacy substream for all notification protocols expect the first one.